### PR TITLE
Clean up E0759 explanation

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC 

cc @pickfire

### CODE REVIEW

File: src/librustc_error_codes/error_codes/E0759.md [hunk 1]
```diff
@@ -5,14 +5,11 @@ Erroneous code examples:
 ```compile_fail,E0759
 use std::fmt::Debug;
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0759.md:6]: Does this still needs `Debug`? I saw that the rest have the `use` removed.

**Comment by pickfire** [comment]
> A `'static` requirement in a return type involving a trait is not fulfilled.

In my humble opinion, I feel like that needs some change to make it easier to read. Maybe?

> Return type involving a trait did not require `'static`.

File: src/librustc_error_codes/error_codes/E0759.md [hunk 2]
```diff
@@ -21,14 +18,11 @@ These examples have the same semantics as the following:
 
 ```compile_fail,E0759
 # use std::fmt::Debug;
```
Comment (pickfire) [inline src/librustc_error_codes/error_codes/E0759.md:20]: Same goes for this.

**Comment by pickfire** [comment]
> These examples have the same semantics as the following:

I feel like that should be changed to something like. I don't understand how "same semantics" is part of the context, I can only see that it is a fix for the above code because of the `ok!` comment.

> Add `'static` requirement to fix it:

File: src/librustc_error_codes/error_codes/E0759.md (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustc_error_codes/error_codes/E0759.md:6]: ;)

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by pickfire — APPROVED** [review]
Looks good to me.

**Pull request merged**